rideshare.rb

```diff
@@ -0,0 +1,50 @@
+
+ride_share = {
+  DR0001: [
+    {cost: 10, date: "3rd Feb 2016", rider_id: "RD0003", rating: 3}, 
+    {cost: 45, date: "5th Feb 2016", rider_id: "RD0003", rating: 2}, 
+    {cost: 30, date: "3rd Feb 2016", rider_id: "RD0015", rating: 4}
+  ],
+  DR0002: [
+    {cost: 35, date: "5th Feb 2016", rider_id: "RD0066", rating: 3}, 
+    {cost: 25, date: "3rd Feb 2016", rider_id: "RD0073", rating: 5}, 
+    {cost: 15, date: "4th Feb 2016", rider_id: "RD0013", rating: 1}
+  ],
+  DR0003: [
+    {cost: 50, date: "5th Feb 2016", rider_id: "RD0003", rating: 2}, 
+    {cost: 5, date: "4th Feb 2016", rider_id: "RD0066", rating: 5}
+  ], 
+  DR0004: [
+    {cost: 5, date: "3rd Feb 2016", rider_id: "RD0022", rating: 5}, 
+    {cost: 20, date: "5th Feb 2016", rider_id: "RD0073", rating: 5}, 
+    {cost: 10, date: "4th Feb 2016", rider_id: "RD0022", rating: 4}
+  ]
+}
+
+max_cost = 0
+most_cost_driver = 0
+highest_rating = 0
+highest_rated_driver = 0
+
+ride_share.each do |driver, ride|
```

Comment:
This is clear and efficient code that makes use of one for each loop that loops through your data structure and performs all the necessary computations. Nice work!